src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,454 @@
+         *
+         * @param[in] registry  Registry object pointing to where the data is.
+         *
+         * @return Return package location.
+         */
+        const std::string getLocation(const Utils::Registry& registry)
+        {
+            std::string location;
+
+            return registry.string("PackageRootFolder", location) ? location : UNKNOWN_VALUE;
+        }
+
+
+        const std::string getInstallTime(const struct attributesAppName& attributes)
+        {
+            std::string installTime { UNKNOWN_VALUE };
+            unsigned long long int value;
+
+            // Name of main registry is: app_Name + _ + app_UUID
+            const std::string mainRegistry {attributes.name + "_" + attributes.uuid};
+
+            try
+            {
+                const Utils::Registry installTimeRegistry {Utils::Registry(m_key, m_userId + "\\" + APPLICATION_INSTALL_TIME_REGISTRY + "\\" + mainRegistry + "\\" + m_appName)};
+
+                if (installTimeRegistry.longint("InstallTime", value))
+                {
+
+                    // Format of value is 18-digit LDAP/FILETIME timestamps.
+                    // 18-digit LDAP/FILETIME timestamps -> Epoch/Unix time
+                    // (value/10000000ULL) - 11644473600ULL
+                    const time_t time {static_cast<long int>((value / 10000000ULL) - WINDOWS_UNIX_EPOCH_DIFF_SECONDS)};
+                    char formatString[20] = {0};
+
+                    std::strftime(formatString, sizeof(formatString), "%Y/%m/%d %H:%M:%S", std::localtime(&time));
+                    installTime.assign(formatString);
+                }
+            }
+            catch (...)
+            {
+            }
+
+            return installTime;
+        }
+
+        /*
+         * @brief Get application name.
+         *
+         * @param[in] fullName Full application name.
+         * @param[in] registry Registry object pointing to where the data is located.
+         *
+         * @return Return name application.
+         */
+        const std::string getName(const std::string& fullName, const Utils::Registry& registry)
+        {
+            std::string name;
+
+            /*
+             * Example of  fullName
+             *
+             * Microsoft.Windows.Search
+             * Microsoft.SkypeApp
+             */
+            const auto keyName { Utils::split(fullName, '.').back() }; // Will only use the last vector element.
+            constexpr auto INVALID_NAME_APP { "@{" };
+
+            try
+            {
+                for (const auto& folder : registry.enumerate())
+                {
+                    std::string value;
+                    const Utils::Registry nameReg(m_key, m_userId  + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_appName + "\\" + folder + "\\Capabilities");
+
+                    if (nameReg.string("ApplicationName", value))
+                    {
+                        name = value;
```

Comment:
Could a break be added here?

Reply:
Hi @Dwordcito 

Yes, if we can get the name we no longer need to continue with the `for` loop.